overlord/snapstate/backend/link.go

```diff
@@ -47,6 +47,10 @@ type LinkContext struct {
 	// RunInhibitHint is used only in Unlink snap, and can be used to
 	// establish run inhibition lock for refresh operations.
 	RunInhibitHint runinhibit.Hint
+
+	// RequireSnapdTooling indicates that the apps and services generated
+	// when linking need to use the snapd tooling.
+	RequireSnapdTooling bool
```

Comment:
the issue is really that snapd snap is a separate mount.  Maybe RequireMountedSnapdSnap ?